ui/app.go

```diff
@@ -71,16 +71,29 @@ func (app *App) onInputCapture(e *tcell.EventKey) *tcell.EventKey {
 
 	if app.MainFlex.GetItemCount() != 0 {
 		switch e.Name() {
+		// Avoid focusing what is already focused so that the widgets can use shortcuts
 		case app.Config.Keys.ToggleGuildsList:
+			if app.GuildsList.HasFocus() {
+				return e
+			}
 			app.SetFocus(app.GuildsList)
 			return nil
 		case app.Config.Keys.ToggleChannelsTreeView:
+			if app.ChannelsTreeView.HasFocus() {
+				return e
+			}
 			app.SetFocus(app.ChannelsTreeView)
 			return nil
 		case app.Config.Keys.ToggleMessagesTextView:
+			if app.ChannelsTreeView.HasFocus() {
+				return e
+			}
 			app.SetFocus(app.MessagesTextView)
 			return nil
 		case app.Config.Keys.ToggleMessageInputField:
+			if app.MessageInputField.HasFocus() {
+				return e
+			}
```

Comment:
I think these if-statements are unnecessarily making the code longer and ugly.

Reply:
Well I thought on these ifs because the default "g" for guild conflicts with the vi-like vi to go up, but i guess the right solution is have the user remap it anyway.